src/pages/students/[student_id].tsx

```diff
@@ -16,16 +16,36 @@ const ViewStudentPage = () => {
     { enabled: Boolean(student_id) }
   );
 
-  const { mutate } = trpc.student.unassignStudent.useMutation();
+  const { data: ieps } = trpc.student.getIeps.useQuery({
+    student_id: student_id as string,
+  });
 
+  const archiveMutation = trpc.student.unassignStudent.useMutation();
   const archiveStudent = async () => {
     if (!student) {
       return;
     }
-    mutate({ student_id: student.student_id });
+    archiveMutation.mutate({ student_id: student.student_id });
```

Comment:
discussed this with @hieungo89, there's two mutation APIs in React Query:
- implicit async (used here): `.mutate()` will return `void` immediately and other parts of the return value of `.useMutation()` can be used to display success or error states. Super easy to use when your UI can display the current state.
- explicit async: `.mutateAsync()` will return `Promise<void>` and either resolve or throw like a normal Promise when the mutation completes. This is better to use for control flow than `.mutate()`, as currently in the snippet above the user will be redirected no matter what.

```suggestion
    await archiveMutation.mutateAsync({ student_id: student.student_id });
```

Reply:
Done